source/ota_mqtt.c

```diff
@@ -292,21 +300,17 @@ static OtaErr_t unsubscribeFromDataStream( const OtaAgentContext_t * pAgentCtx )
 
     /* This buffer is used to store the generated MQTT topic. The static size
      * is calculated from the template and the corresponding parameters. */
-    const uint32_t rxStreamBufferSize =
-        CONST_STRLEN( pOtaStreamDataTopicTemplate )
-        + otaconfigMAX_THINGNAME_LEN
-        + STREAM_NAME_MAX_LEN
-        + NULL_CHAR_LEN;
-    char pOtaRxStreamTopic[ rxStreamBufferSize ];
+    char pOtaRxStreamTopic[ TOPIC_STREAM_DATA_BUFFER_SIZE ];
     uint16_t topicLen = 0;
-    const OtaFileContext_t * pFileContext = 0;
+    const OtaFileContext_t * pFileContext = NULL;
 
     assert( pAgentCtx != NULL );
 
     pFileContext = &( pAgentCtx->fileContext );
 
     /* Try to build the dynamic data stream topic and unsubscribe from it. */
-    topicLen = ( uint16_t ) snprintf( pOtaRxStreamTopic, /*lint -e586 Intentionally using snprintf. */
+    /* coverity[misra_c_2012_rule_21_6_violation] Buffer overlap will never happen. */
```

Comment:
Why use snprintf()?  Seems like you could use strncpy() instead.

Reply:
Because we're not just handling strings, see here, https://github.com/aws/ota-for-aws-iot-embedded-sdk/blob/b76149ae0737a04cc48fa99b79a9257211e0e8c1/source/ota_mqtt.c#L63-L75. Using snprintf would handle all cases, and we know the undefined behavior can never happen here.